src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -301,10 +317,16 @@ private static void validateGetItemKeysAndAttribute(KeysAndAttributes keysAndAtt
      */
     @Override
     public CreateTableResult createTable(CreateTableRequest createTableRequest) {
-        tableMappingFactory.validateCreateVirtualTableRequest(createTableRequest);
-        // TODO: persist virtual to physical secondary index map in repo
-        return new CreateTableResult()
-            .withTableDescription(withTenantStreamArn(mtTableDescriptionRepo.createTable(createTableRequest)));
+        return createTable(createTableRequest, false);
+    }
+
+    private CreateTableResult createTable(CreateTableRequest createTableRequest, boolean isMultitenant) {
+        TableMapping tableMapping = tableMappingFactory.createNewTableMapping(createTableRequest, isMultitenant);
```

Comment:
TableMappingFactory.createNewTableMapping() validates the request and creates the physical table if it doesn't already exist.